### DISCUSSION TITLE
Is there a way in (Next 10.x) to structure test files inside pages directory?

### DISCUSSION THREAD

**Discussion by piotrrussw** [vercel/next.js#21467] [Help]
Hi,

I have quite a large next.js project which is structured in a "folder-by-feature", so a Component has all related files close together, e.g.
```
About/
- About.js
- About.module.css
- About.test.js
```

I would like to be consistent so inside `page/` directory I would like to also have the same structure. But I found no way to exclude test files from being treated as a route. I searched lots of old issues but none of them has a decent solution, also non of the given workarounds works in next 10.x
Is there a 2021 solution for this approach? Or any plans to exclude test files from pages?

**Comment by rokinsky** [comment]
You can use `pages` directory only for routing and just re-export needed functions, i.e.

```js
// pages/about.js
export { default, getStaticProps } from "../Components/About/About";
```

then in `Components/About` keep css and spec files that are not related to routing.

**Reply by piotrrussw** [reply]
Thanks, It could be a solution, but I am not at the time where I feel comfortable changing it. So I'd rather find a way to tell next to ignore this `*.test.js` in pages directory from being routes.

**Reply by rokinsky** [reply]
Unfortunately, this isn't possible at the moment.

**Reply by piotrrussw** [reply]
Ok, thanks for the answer

**Reply by benjaminpearson** [reply]
You could use https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions to say ".page.js" are pages and then you could have components and tests in the folder. I haven't tried it on a project yet but am also keep to folders by feature rather than folders by type.

**Comment by tarunama** [comment]
> Is there a 2021 solution for this approach?

Use `jest` or `Cypress`.
